src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/NodeMappingToURIsImpl.java

```diff
@@ -21,7 +21,7 @@ public Node map(final LPGNode node) {
     public LPGNode unmap(final Node node) {
         if (!node.isURI())
             throw new IllegalArgumentException("The given RDF term (" + node.toString() + ") is not an URI node.");
-        if (!node.getURI().startsWith(NSNODE))
+        if (!isPossibleResult(node))
```

Comment:
Same comment here. Please remove the first if statement here as well (and adjust the error message).